app/components/certificate/certificate-request.tsx

```diff
@@ -0,0 +1,51 @@
+import { Flex, Heading, Text, HStack, Button } from '@chakra-ui/react';
+
+import Description from './description';
+
+interface CertificateRequestProps {
+  subject: string;
+  validFrom: Date;
+  validTo: Date;
+  setLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setCertificateRequested: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export default function CertificateRequestView({
+  subject,
+  validFrom,
+  validTo,
+  setLoading,
+  setCertificateRequested,
+}: CertificateRequestProps) {
+  function handleSubmit() {
+    setLoading(true);
+    setTimeout(() => {
+      setCertificateRequested(true);
+      setLoading(false);
+    }, 5000);
+  }
+
+  return (
+    <Flex flexDirection="column" gap="5" width="4xl">
+      <Description
+        description="Initial: Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s"
+        certRequested={false}
+        validFrom={validFrom}
+        validTo={validTo}
+      />
+      <Flex flexDirection="column" gap="1">
+        <Heading as="h3" size="md" marginBottom="3">
+          Domain Name
+        </Heading>
+        <Text>Lorem Ipsum is simply dummy text of the printing and typesetting industry</Text>
+
+        <HStack>
+          <Text fontWeight="bold">Domain Name:</Text> <Text>{subject}</Text>
+        </HStack>
+      </Flex>
+      <Button width="3xs" shadow="xl" onClick={handleSubmit}>
```

Comment:
This `onClick` should call into the function passed down on props, and not bother doing anything in this component (put logic up the component tree).

Reply:
I was a bit reluctant, because I'm not quite familiar with how we handle the data types for functions. Should be working on the next commit